### ISSUE TITLE
Name-based search not handling modifiers correctly

### ISSUE THREAD

**Issue by jensschuppe** [Project60/org.project60.banking#206]
The `BankingLookup.contactbyname` API call does not apply modifiers correctly due to typos: Instead of comparing the modifier mode, it sets it to "entire" for all modifiers which results in replacing and e.g. "alternative" mode not being able anymore to search for the original value. Just to illustrate those hard-to-debug typo issues, here's an example scenario:

Search string: `Mr. John Doess`

Default matcher configuration (excerpt):
```JSON
"modifiers": [
      {
        "search": ":ss:",
        "replace": "ß",
        "mode": "alternative"
      }
    ]
```

Faulty condition when checking for modifier mode:

```PHP
if (!empty($modifier['search']) && !empty($modifier['replace']) && (empty($modifier['mode']) || $modifier['mode'] = 'entire') )
```
Note the `=` instead of `==` in the last condition which sets the mode to "entire" and causes a replace overwriting the original value: The search string becomes `mr. john doeß`.

Some lines further down:
```PHP
if (!empty($modifier['search']) && !empty($modifier['replace']) && !empty($modifier['mode']) && $modifier['mode'] = 'alternative' )
```
Same issue: `=` instead of `==` - this causes the check to be always done, since the assignment results in `TRUE`, but as the search string got overridden, there will be no match for `Doess` e.g. as part of a contact name. However, contacts named `Doeß` will still be matched.

**Comment by jensschuppe** [comment]
See PR #207.

**Comment by systopia** [comment]
Will be released with ``0.7``
